packages/dendron-cli/src/commands/notes.ts

```diff
@@ -341,11 +342,12 @@ export class NoteCLICommand extends CLICommand<CommandOpts, CommandOutput> {
         }
         case NoteCommands.DELETE: {
           const { query, vault } = checkQueryAndVault(opts);
-          const note = NoteUtils.getNoteByFnameFromEngine({
-            fname: query,
-            vault,
-            engine,
-          });
+          const note = (
+            await engine.findNotes({
```

Comment:
makes me think we should have a `findUnique` method for these cases like [prisma](https://www.prisma.io/docs/reference/api-reference/prisma-client-reference#findunique). maybe future iteration

Reply:
`findUnique` or `findFirst` looks tempting. I was concerned about polluting the api with too many methods since we already have findNote and findNoteMeta. Perhaps when we refactor the clients it'll give us a chance to revisit these

Reply:
isn't that the same as `getNote`?

Reply:
`getNote` currently just supports getting by note id. Are you suggesting we overload it to pass it 2 different sets of parameters `id || (fname + vault)`?

Reply:
Er, I guess I'm just confused.  If `findUnique` takes fname + vault, what's the difference between that and `findNote()[0]` or something similar. 

Seems like a convenience method, but I don't _think_ we should be trying to do stuff like `findNote()[0]` if possible.

Reply:
The main use case looks to be a convenience method. Secondary use cases could be added verification to ensure that fname + vault is unique and if there are multiple matches, `findUnique` can throw an error